api/types/types.proto

```diff
@@ -2624,6 +2634,90 @@ message GithubConnectorSpecV3 {
     string Display = 5 [ (gogoproto.jsontag) = "display" ];
 }
 
+// SAMLAttributesToRolesWarnings contains warnings produced during the process of mapping the SAML
+// attributes to roles.
+message SAMLAttributesToRolesWarnings {
+    // Message is main user-facing message to be shown.
+    string Message = 1 [ (gogoproto.jsontag) = "message,omitempty" ];
+    // Warnings is a set of distinct warnings to be reported.
+    repeated string Warnings = 2 [ (gogoproto.jsontag) = "warnings,omitempty" ];
+}
+
+// CreateUserParams represents the user creation parameters as called during SSO login flow.
+message CreateUserParams {
+    // ConnectorName is the name of the connector used for SSO login flow.
+    string ConnectorName = 1 [ (gogoproto.jsontag) = "connector_name,omitempty" ];
+    // Username is the name of the user to be created.
+    string Username = 2 [ (gogoproto.jsontag) = "username,omitempty" ];
+    // Logins is a list of available unix logins.
+    repeated string Logins = 3 [ (gogoproto.jsontag) = "logins,omitempty" ];
+    // KubeGroups is a list of assigned kube groups.
+    repeated string KubeGroups = 4 [ (gogoproto.jsontag) = "kube_groups,omitempty" ];
+    // KubeUsers is a list of available kube users.
+    repeated string KubeUsers = 5 [ (gogoproto.jsontag) = "kube_users,omitempty" ];
```

Comment:
Do we need to include database users/names here also?

Reply:
No, those are in `Traits` field. `KubeUsers` and `KubeGroups` are something of a special (legacy) case, I expect these to go away at some point in the future.